_includes/docs/user-guide/firmware.md

```diff
@@ -42,10 +58,84 @@ Deletion of the firmware packages that is assigned to at least one device or dev
+any boot issues, and which ones have already been updated.
+It is created automatically when you add firmware to ThingsBoard.
+There you can see a list of all devices with full information about their firmware. 
+
+{% include images-gallery.html imageCollection="fw-dashboard" %}
+
+##### Firmware status
+
+Click the "History firmware update" button next to the device name to learn about the firmware update status of specific device.
+
+{% include images-gallery.html imageCollection="fw-status" %}
+
+The possible state transition can be: QUEUED -> INITIATED -> DOWNLOADING -> DOWNLOADED -> VERIFIED -> UPDATING -> UPDATED/FAILED. 
+The system uses the following algorithm:
+1. Firmware was created.
+2. Telemetry (target_fw_title, target_fw_verison and fw_state(QUEUED)) is sent on the device.
+3. Device receives notification about firmware update and saves attributes. Fw_states changes to INITIATED.
+4. Transport firmware client gets attribute update.
+5. If the firmware is not the same as the previous one, transport firmware client requests the new firmware
+   and sends telemetry with fw_state DOWNLOADING.
+6. Client gets all chunks and sends telemetry fw_state DOWNLOADED.
+7. Client checks firmware and sends telemetry fw_state VERIFIED.
+8. If verification is successful, client will send telemetry fw_state UPDATED. If it's not successful, the telemetry will be fw_state FAILED.
+
+So, there are eight statuses in total. Two of them are created by us and all the rest are generated by the device itself.
+
+1. _QUEUED_ - notification is in a queue to be sent to the device. By default, in the queue can be up to 100 devices. [It can be configured](/docs/{{docsPrefix}}user-guide/firmware/?remoteintegrationdockerinstall=http#thingsboard-configuration-file).
+2. _INITIATED_ - telemetry was sent and device was notified. "Device waiting" tab on the Firmware dashboard. 
+3. _DOWNLOADING_ - telemetry was received and device started downloading firmware.
+4. _DOWNLOADED_ - device completed downloading of the firmware.
+5. _VERIFIED_ - if the checksum of received data was successfully verified by the device.
+6. _UPDATING_ - if the verification is successful, the device firmware is updating. "Device updating" tab on the Firmware dashboard.
+7. _UPDATED_ - the firmware was updated to the next version. When fw_state changes to UPDATED, the system updates current_fw_title and current_fw_version to the new values. "Device updated" tab on the Firmware dashboard.
+8. _FAILED_ - checksum wasn't verified, or the device failed to update. "Device failed" tab on the Firmware dashboard.
+
+{% include images-gallery.html imageCollection="fw-status-1" %}
+
 ## Configuration
 
-TODO: talk about max size setting
\ No newline at end of file
+##### ThingsBoard configuration file
+
+Execute following command to edit ThingsBoard configuration file:
```

Comment:
Add an explanation why user may want to configure this parameters. Ask @YevhenBondarenko for comments